x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/hooks/use_create_analytics_form/use_create_analytics_form.ts

```diff
@@ -90,9 +91,13 @@ export const useCreateAnalyticsForm = (): CreateAnalyticsFormProps => {
     resetRequestMessages();
     setIsModalButtonDisabled(true);
 
-    const analyticsJobConfig = isAdvancedEditorEnabled
+    const analyticsJobConfig = (isAdvancedEditorEnabled
       ? jobConfig
-      : getJobConfigFromFormState(form);
+      : getJobConfigFromFormState(form)) as DataFrameAnalyticsConfig;
```

Comment:
Set as `DataFrameAnalyticsConfig` as that type for the analysis config is still wrapped in a `DeepPartial` at this point but since it has passed validation we don't need that as it's complete.